pages/feed/index.tsx

```diff
@@ -34,15 +43,68 @@ const Feeds = () => {
     });
 
   useEffect(() => {
+    async function getSingleHighlight() {
```

Comment:
The hook approach is ignored because this call needs to be fired only when we have an id in the `router.query` object... Hooks can't be called conditionally

Reply:
Yes, but you can conditionally pass the URL to useSWR if the id is available. If you pass a `null` value, it won't make the request.

Similar to here

https://github.com/open-sauced/insights/blob/beta/lib/hooks/useFetchUser.ts#L7

Make the `id` an optional parameter to the hook

Reply:
Great.. Didn't know this before :)

Reply:
I'm still having same issue with the hook even when i made the id optional. It keep making calls with undefined  even when the id is present